Shared/Samples/Navigate route with rerouting/NavigateRouteWithReroutingView.Model.swift

```diff
@@ -0,0 +1,304 @@
+                let nextManeuverIndex = status.currentManeuverIndex + 1
+                if route.directionManeuvers.indices.contains(nextManeuverIndex) {
+                    let nextDirection = route.directionManeuvers[nextManeuverIndex].text
+                    statusMessage.append("\nNext direction: \(nextDirection)")
+                }
+                
+            case .reached:
+                if status.remainingDestinationCount > 1 {
+                    statusMessage = "Intermediate stop reached, continue to next stop."
+                    try? await routeTracker.switchToNextDestination()
+                } else {
+                    await locationDisplay.dataSource.stop()
+                    statusMessage = "Destination reached."
+                }
+                
+            @unknown default:
+                break
+            }
+        }
+        
+        /// Speaks a given voice guidance.
+        /// - Parameter voiceGuidance: The `VoiceGuidance`.
+        func speakVoiceGuidance(_ voiceGuidance: VoiceGuidance) {
+            guard !voiceGuidance.text.isEmpty else { return }
+            
+            let utterance = AVSpeechUtterance(string: voiceGuidance.text)
+            speechSynthesizer.stopSpeaking(at: .word)
+            speechSynthesizer.speak(utterance)
+        }
+        
+        /// Initializes the location display, route graphic, and viewpoint.
+        private func initializeProgressTracking() {
+            // Create a route tracker location data source to snap the location display to the route.
+            let routeTrackerLocationDataSource = RouteTrackerLocationDataSource(
+                routeTracker: routeTracker,
+                locationDataSource: simulatedDataSource
+            )
+            
+            // Set location display's data source.
+            locationDisplay.dataSource = routeTrackerLocationDataSource
+            
+            // Update the remaining route graphic and center the map's viewpoint on it.
+            guard let routeGeometry = route.geometry else { return }
+            remainingRouteGraphic.geometry = routeGeometry
+            viewpoint = Viewpoint(center: routeGeometry.extent.center, scale: 23e3, rotation: 0)
+        }
+        
+        /// Makes a route tracker that supports rerouting.
+        /// - Parameters:
+        ///   - routeResult: A `RouteResult` generated from a route task solve.
+        ///   - routeTask: The `RouteTask` used to generate the route result.
+        ///   - routeParameters: The `RouteParameters` used to solve for the route result.
+        /// - Returns: A new `RouteTracker`.
+        private func makeRouteTracker(
+            routeResult: RouteResult,
+            routeTask: RouteTask,
+            routeParameters: RouteParameters
+        ) async throws -> RouteTracker? {
+            // Make the route tracker from the route result.
+            let routeTracker = RouteTracker(
```

Comment:
The route tracker initializer can be force unwrapped, IMO. As we know the route result's first route would always be valid for the sample.